doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,183 @@
-current buffer, using the *Edit/External Editor* menu.
-
-Interactive navigation into Coq scripts
---------------------------------------------
-
-The running buffer is the one where navigation takes place. The toolbar offers
-five basic commands for this. The first one, represented by a down arrow icon,
-is for going forward executing one command. If that command is successful, the
-part of the script that has been executed is displayed on a background with the
-processed color. If that command fails, the error message is displayed in the
-message window, and the location of the error is emphasized by an underline in
-the error foreground color (red by default).
-
-In the figure :ref:`CoqIDE main screen <coqide_mainscreen>`,
+-----------------------------------------
+
+The *File* menu lets you open files, create new buffers, save buffers to a file,
+and print or export them in various formats.
+
+Text editing provides the basic operations such as copy, cut, paste, find and replace.
+Most editing operations are shown in the *Edit* menu.  Most menu entries show the equivalent
+keystrokes to perform the action.  If you need more complex editor commands, you can
+launch your favorite text editor on the current buffer, using the
+*Edit/External Editor* menu. When you're done editing, you must currently must
+reopen the file to see your changes.  Also note:
+
+- Undo is also available as Ctrl-Z (Command-Z on macOS).  Redo is Ctrl-Shift-Z
+  (Shift-Command-Z on macOS).
+- Select all is Ctrl-A (Command-A on macOS).
+- Ctrl-Delete deletes a word of text after the cursor
+- Ctrl-Backspace deletes a word of text before the cursor
+- Commenting and uncommenting the current line or selected text is available in
+  the *Tools* menu.  If some text is selected, exactly that text is commented out;
+  otherwise the line containing the cursor is commented out.  To uncomment, position
+  the cursor between `(*` and `*)` or select any text between them.
+
+Files are automatically saved periodically to a recovery file.  For example,
+`foo.v` is saved to `#foo.v#` every 10 seconds by default.  You can change the
+interval in the *Edit / Preferences / Files* dialog.
+
+In the *View* menu, you can set several printing options that
+correspond to options that can appear in the script.  For example, "Display
+notations" on the menu corresponds to the :flag:`Printing Notations` flag.  You
+should use the menu instead of controlling these settings in your script.
+
+Running Coq scripts
+-------------------
+
+Operations for running the script are available in the *Navigation* menu,
+from the toolbar and from the keyboard.  These include:
+
+- Forward (Ctrl-Down) to run one command or tactic
+- Backward (Ctrl-Up) undo one command or tactic
+- Run to cursor (Ctrl-Right) to run commands up to the cursor
+- Run to end (Ctrl-End) to run commands to the end of the buffer
+- Reset Coq (Ctrl-Home) to restart the Coq process
+- Interrupt to stop processing commands
```

Comment:
In my experience, if CoqIDE is very busy, for example printing a huge term, interrupt can be somewhat unresponsive. The goal is to not undo the processed part of the script and just interrupt the current command. If CoqIDE is really not responsive however, resetting Coq with Ctrl-Home typically does the job.

I don't know if this needs mentioning.

Reply:
Interrupt is implemented with a signal sent directly to coqidetop that sets a flag.  The interpreter checks the flag before starting the next command.  I tweaked the wording.